src/api/define.js

```diff
@@ -122,13 +133,16 @@ export default function (name, opts) {
   Ctor[$props] = createInitProps(Ctor);
   Ctor[$renderer] = createRenderer(Ctor);
 
-  if (customElementsV1) {
-    window.customElements.define(uniqueName, Ctor, { extends: Ctor.extends });
-  } else if (customElementsV0) {
-    return document.registerElement(uniqueName, Ctor);
-  } else {
-    throw new Error('Skate requires native custom element support or a polyfill.');
+  let res;
+  try {
```

Comment:
You don't need to `try` in v1; you can just check if it exists using `window.customElements.get()`. Admittedly this kinda messes up the clean logic you've laid out, though. Maybe you can split out registration into `registerV0` and `registerV1` or something?